flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/pattern/conditions/SimpleCondition.java

```diff
@@ -40,4 +40,13 @@
     public boolean filter(T value, Context<T> ctx) throws Exception {
         return filter(value);
     }
+
+    public static <T> SimpleCondition<T> of(FilterFunction<T> filters) {
```

Comment:
Alternatively we could deprecate this entire class and add `Pattern#where`/`or` variations that accept a `FilterFunction`.

Reply:
@zentol, IMO, it's better to deprecate the `SimpleCondition` class and replace the `Condition` with the `FilterFunction`.

Reply:
At the end of the day this doesn't need to be the final solution.

While `Pattern#where(FilterFunction)` is a lot more user-friendly this would be a breaking change, because an existing `SimpleCondition` would satisfy both signatures which the compiler rejects.
I'd like to avoid using different method names for the `FilterFunction` variants.

We could also modify `SimpleCondition` to not implement the `FilterFunction` interface, but this would again be a potentially breaking change.

Reply:
This is already a big improvement in usability, and I don't see how to do better without introducing a breaking (or confusing) change.